### PR TITLE
[DEVOPS-252] Tide Workflows

### PR DESCRIPTION
**Pull request by FleetAdmiralButter**
1) Includes the build, end-to-end, API, and UAT workflows for Tide projects.

### CODE REVIEW

File: .github/workflows/api.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Run core Newman API tests
+        run: |
+          cd /app
+          if [ -e tests/newman/tide.collection.json ]
+          then
+            if [[ ${TEST_NEWMAN_DISABLE_CORE} == true ]]
+            then
+              echo "==> Core Tests are disabled by env var TEST_NEWMAN_DISABLE_CORE. Skip the test."
+            else
+              case ${CIRCLE_BRANCH} in
```
Comment (tim-yao) [inline .github/workflows/api.yml:56]: Hi @FleetAdmiralButter `CIRCLE_BRANCH` is not available, we need to replace it by GITHUB vars

File: .github/workflows/api.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+name: api
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: SDP testing Docker image tag
+        type: string
+        required: false
+        default: migration
```
Comment (tim-yao) [inline .github/workflows/api.yml:10]: @FleetAdmiralButter This is not used, no need to add it

File: .github/workflows/api.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        type: string
+        required: true
+      disable_core:
+        description: Set the TEST_NEWMAN_DISABLE_CORE env var
+        type: boolean
+        required: false
+        default: false
+
+env:
+  TEST_NEWMAN_DISABLE_CORE: ${{ inputs.disable_core }}
```
Comment (tim-yao) [inline .github/workflows/api.yml:22]: Hi @FleetAdmiralButter this was managed by project-level env vars, so we can turn off the API test if we don't want it. I guess we can continue to do it, it won't be controlled by workflow inputs.

File: .github/workflows/api.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        if: always()
+        with:
+          name: core-test-report
+          path: /app/report/tide.collection.report.html
+      - name: Upload custom API test report
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: custom-test-report
+          path: /app/report/custom.collection.report.html
```
Comment (tim-yao) [inline .github/workflows/api.yml:91]: Missing the notify_slack step

**Review by tim-yao — APPROVED** [review]

**Review by steveworley — APPROVED** [review]

**Pull request merged**